src/config.cpp

```diff
@@ -62,6 +62,19 @@ struct ProviderOptionsArray_Element : JSON::Element {
   ProviderOptionsObject_Element object_{v_};
 };
 
+GraphOptimizationLevel getGraphOptimizationLevel(std::string_view name) {
+  if (name =="ORT_DISABLE_ALL") {
+      return ORT_DISABLE_ALL;
+  } else if (name == "ORT_ENABLE_BASIC") {
+      return ORT_ENABLE_BASIC;
+  } else if (name == "ORT_ENABLE_EXTENDED") {
+      return ORT_ENABLE_EXTENDED;
+  } else if (name == "ORT_ENABLE_ALL") {
+      return ORT_ENABLE_ALL;
+  } else
+    throw JSON::unknown_value_error{};
```

Comment:
 ```suggestion
    throw std::runtime_error("Invalid graph_optimization_level value:" + name);
```

This exception isn't what you want, it's to say that the value name is unknown (it'll say "graph_optimization_level" is not recognized). The value name is recognized, but it's being set to an unrecognized value. This should be more useful: